internal-messaging/messaging-service/cc/MessageSenderInterface.cc

```diff
@@ -12,13 +12,27 @@ void MessageSenderInterface::SetRecipient(unsigned int recipient){
     client_socket_ = UnixDomainStreamSocketClient(path);
 }
 
+//Send message to socket
 void MessageSenderInterface::SendFlattenedMessage(char message[]) {
     client_socket_.Send(message);
 }
 
-void MessageSenderInterface::SendDataMessage(DataMessage message) {
+//Send message to socket and await reply
+string MessageSenderInterface::SendFlattenedMessageAwaitReply(char message[]) {
```

Comment:
This seems dangerous due to the [Two General's Problem](https://en.wikipedia.org/wiki/Two_Generals%27_Problem). There should be a limit to how long this waits for (this might be implemented in `SendMessageAwaitReply`, just haven't seen it yet)

Reply:
Additionally, sending a message like this should include a flag that indicates that the OS is awaiting a reply.

Reply:
Reply flag added as new feature to be implemented